openrtb_ext/bidders.go

```diff
@@ -15,6 +15,20 @@ import (
 // BidderName refers to a core bidder id or an alias id.
 type BidderName string
 
+var aliasBidderNames []BidderName
+
+func SetAliasBidderName(name BidderName) {
+	aliasBidderNames = append(aliasBidderNames, name)
+}
+
+func GetAliasBidderNamesSet() map[BidderName]bool {
+	set := map[BidderName]bool{}
+	for _, name := range aliasBidderNames {
+		set[name] = true
+	}
+	return set
+}
```

Comment:
Please help to understand usage of this function? 
1) `set` here is something that should be evaluated once at the server start up, not every time this function is called
2)  `aliasBidderNames` can be a set, I don't see why we should have it as array. 

Is it correct you use set for faster check if element is present:
```go
if _, ok := GetAliasBidderNamesSet()[bidderName]; ok {
	continue
}
```
do you think it would work better if you replace  array to set?
If you only need to check element is present in set you can use struct{} instead of bool. 
Empty struct takes no space in memory.

```go

var aliasBidderNames map[BidderName]struct{} 

func SetAliasBidderName(name BidderName) {
	aliasBidderNames[name] = struct{}{}
}

func GetAliasBidderNamesSet() map[BidderName]struct{} {
	return aliasBidderNames
}

func HasAliasBidderNames(bidderName BidderName) bool {
	_, ok := aliasBidderNames[bidderName]
	return ok
}
```

Reply:
Ok, I see why you need it in array when you add it to coreBidderNames. 
If this is just one time call, then adding conversion function will make sense:
```go
func ToSlice() []BidderName{
	valueSlice := make([]BidderName, 0, len(aliasBidderNames))
	for  key, _ := range aliasBidderNames {
		valueSlice = append(valueSlice, key)
	}
	return valueSlice
}
```

Reply:
Thanks @VeronikaSolovei9 for the review!
Here are my thoughts - 
- Agreed that we should use struct to create a set instead of bool to save on memory space. Made that change in the [subsequent commit.](https://github.com/prebid/prebid-server/pull/3027/commits/3f4c0c716f93c54483ece56a8577b3552ffc21d2)
- On replacing array to set, I think it is still better to keep `aliasBidderNames` as slice. The `GetAliasBidderNamesSet` is being used just [once here](https://github.com/prebid/prebid-server/blob/a9831f28a65a96df8e11810dcaeacc8555d92d61/openrtb_ext/bidders_validate_test.go#L27-L28).
- However, we have usage of coreBidderNames at multiple places where `aliasBidderNames` is being used so it makes more sense to keep `aliasBidderNames` as slice.

Reply:
Ok I see, this makes sense. Thank you.